app/controllers/subscriptions_controller.rb

```diff
@@ -51,6 +53,18 @@ def update
 
   private
 
+  def require_login?
+    true unless action_name == "create" && Rails.env.test?
```

Comment:
we could maybe make this even more specific here by checking that the request is `paypal?`